pkg/util/cache/cache.go

```diff
@@ -46,18 +48,21 @@ type Scavenger[K comparable] interface {
 	//	room for all the pending entries if the thrown entry is evicted. Typically, the collector will get multiple true
 	//	before it gets a false.
 	Spare(key K) func(K) bool
+	Replace(key K) (bool, func(K) bool)
 }
 
 type LazyScavenger[K comparable] struct {
 	capacity int64
 	size     int64
 	weight   func(K) int64
+	weights  map[K]int64
```

Comment:
Since the weight calculation is cheap, it would be better if we skip caching the result. Caching brings complexity IMO, since we have to deal with the concurrent key update stuffs.

Reply:
> Since the weight calculation is cheap, it would be better if we skip caching the result. Caching brings complexity IMO, since we have to deal with the concurrent key update stuffs.

because update index operation will update the segment's loadinfo, and the weight function is calculated using the loadinfo. If we dont save the weights of previous loadinfo, we will lose it and the size will be imprecise.